core/controllers/suggestion_test.py

```diff
@@ -188,6 +190,210 @@ def test_create_suggestion(self):
+                suggestion_to_accept['suggestion_id']),
+            {}, csrf_token=csrf_token, expected_status_int=400)
+
+        self.assertEqual(
+            response['error'],
+            'The exploration id provided does not match the exploration id '
+            'present as part of the suggestion_id')
+
+        self.logout()
+
+    def test_owner_of_exploration_cannot_repond_to_own_suggestion(self):
+        self.login(self.EDITOR_EMAIL)
+
+        thread_id = feedback_services.create_thread(
+            suggestion_models.TARGET_TYPE_EXPLORATION, self.EXP_ID,
+            self.editor_id, 'description', '', has_suggestion=True)
+
+        suggestion_models.GeneralSuggestionModel.create(
+            suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT,
+            suggestion_models.TARGET_TYPE_EXPLORATION, self.EXP_ID,
+            1, suggestion_models.STATUS_IN_REVIEW, self.editor_id,
+            None, {
+                'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                'property_name': exp_domain.STATE_PROPERTY_CONTENT,
+                'state_name': 'Welcome!',
+                'new_value': 'new_content'
+            }, 'score_category', thread_id)
+
+        response = self.get_html_response('/explore/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        response = self.put_json(
+            '%s/exploration/%s/%s' % (
+                feconf.SUGGESTION_ACTION_URL_PREFIX, self.EXP_ID, thread_id),
+            {}, csrf_token=csrf_token, expected_status_int=401)
+
+        self.assertEqual(
+            response['error'], 'You cannot accept/reject your own suggestion.')
+
+        self.logout()
+
+    def test_suggestion_to_exploration_handler_with_invalid_action(self):
+        self.login(self.EDITOR_EMAIL)
+
+        response = self.get_html_response('/explore/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        suggestion_to_accept = self.get_json(
+            '%s?author_id=%s' % (
+                feconf.SUGGESTION_LIST_URL_PREFIX,
+                self.author_id))['suggestions'][0]
+
+        response = self.get_html_response('/explore/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        response = self.put_json(
+            '%s/exploration/%s/%s' % (
+                feconf.SUGGESTION_ACTION_URL_PREFIX,
+                suggestion_to_accept['target_id'],
+                suggestion_to_accept['suggestion_id']),
+            {'action': 'invalid_action'}, csrf_token=csrf_token,
+            expected_status_int=400)
+
+        self.assertEqual(
+            response['error'], 'Invalid action.')
+
+        self.logout()
+
+    def test_reject_suggestion(self):
+        self.login(self.EDITOR_EMAIL)
+
+        response = self.get_html_response('/explore/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        suggestion_to_accept = self.get_json(
```

Comment:
suggestion_to_reject would be better here...

Reply:
Oops done!